packages/xo-server/src/xapi/index.js

```diff
@@ -1136,7 +1136,13 @@ export default class Xapi extends XapiBase {
     const vifsMap = {}
     if (vm.$pool !== host.$pool) {
       const defaultNetworkRef = find(host.$PIFs, pif => pif.management).$network.$ref
-      for (const vif of vm.$VIFs) {
+      // Add snapshots' VIFs which VM has no VIFs on these devices
+      const vmVifs = vm.$VIFs
+      const vifDevices = new Set(mapToArray(vmVifs, 'device'))
+      const vifs = flatMap(vm.$snapshots, '$VIFs')
+        .filter(vif => !vifDevices.has(vif.device))
```

Comment:
Do we really need to add all snapshots' VIFs? Even if the specific device is already present from an other snapshot's VIF?

Should we need to updater vifDevices while iterating on `vm.snapshots[*].VIFs`?

Reply:
I haven't tested this, but if for each VIF snapshot, the XAPI looks for the VIF VM on the same VIF snapshot device and keeps its network to migrate the VIF snapshot on the same network, we will always have the same error.

Reply:
Let's roll back to the previous, simpler, implementation.